atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -376,6 +377,9 @@ private enum State {
                 timelockService,
                 immutableTimestamp,
                 knowledge);
+        this.transactionMetrics = TransactionMetrics.of(metricsManager.getTaggedRegistry());
+        this.transactionOutcomeMetrics =
+                TransactionOutcomeMetrics.create(transactionMetrics, metricsManager.getTaggedRegistry());
```

Comment:
This is wonky - but TransactionOutcomeMetricsAssert needs to access the TaggedMetricRegistry and I didn't want to make sweeping changes. We could just have the assertThat function take in the registry, which would be a mental model shift in how assertThat is used...